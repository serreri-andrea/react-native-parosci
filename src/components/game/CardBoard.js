/**
 * Created by Andrea on 18/03/2018.
 */

import React, { Component } from 'react';
import {
    StyleSheet,
    Text,
    View,
    FlatList,
    Image,
    TouchableHighlight
} from 'react-native';

import Sizes                from "../../theme/sizes";

export default class CardBoard extends Component{

    constructor(props) {
        super(props);
        this.state = {};
    }

    handleChoosedCard(card){
        this.setState({card:card});
        this.props.callback(card)
    }

    renderCards(item){
        if (item && item.item && item.item.images) {
            return (
                <View>
                    <TouchableHighlight onPress={this.handleChoosedCard.bind(this, item.item.reference)}
                                        underlayColor={'transparent'}>
                        <Image
                            source={this.state.card === item.item.reference ? item.item.images.selected : item.item.images.default}
                            style={{height:Sizes.screen.height / 8, width:Sizes.screen.width / 5, margin:10}}/>
                    </TouchableHighlight>
                </View>
            )
        }else{return(<View/>)}
    }

    render(){
        return(
            <View style={styles.container}>
                <FlatList
                    data={this.props.cards}
                    extraData={this.state}
                    numColumns={3}
                    columnWrapperStyle={styles.container}
                    ItemSeparatorComponent={() => <View style={{margin:0}}/>}
                    renderItem={this.renderCards.bind(this)}
                    keyExtractor={(item, index) => index.toString()}/>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container:{
        marginTop:10,
        marginBottom:10,
        justifyContent:'center',
        alignItems:"center",
    },
});
