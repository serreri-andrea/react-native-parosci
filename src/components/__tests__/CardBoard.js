import 'react-native';
import React            from 'react';
import CardBoard        from '../game/CardBoard';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly: without props', () => {
    const tree = renderer.create(
        <CardBoard/>
    );
    expect(tree).toMatchSnapshot();
});

it('renders correctly: with props', () => {
    const tree = renderer.create(
        <CardBoard cards={"ghjk"} callback={"fghjkl"}/>
    );
    expect(tree).toMatchSnapshot();
});

