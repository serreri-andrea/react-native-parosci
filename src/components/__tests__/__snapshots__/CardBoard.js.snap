// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly: with props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    columnWrapperStyle={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 10,
        "marginTop": 10,
      }
    }
    data="ghjk"
    disableVirtualization={false}
    extraData={Object {}}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    invertStickyHeaders={undefined}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={3}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginBottom": 10,
                "marginTop": 10,
              },
            ]
          }
        >
          <View />
          <View />
          <View />
        </View>
        <View
          style={
            Object {
              "margin": 0,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginBottom": 10,
                "marginTop": 10,
              },
            ]
          }
        >
          <View />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders correctly: without props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    columnWrapperStyle={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 10,
        "marginTop": 10,
      }
    }
    data={undefined}
    disableVirtualization={false}
    extraData={Object {}}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    invertStickyHeaders={undefined}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={3}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
